### ISSUE TITLE
Is quick_tx formally supported for ixgbe?

### ISSUE THREAD

**Issue by morphyno** [appneta/tcpreplay#169]
I just installed 4.1.0 and ran some tests using "quick_tx" mode over ixgbe. I noticed quick_tx is actuall slower than regular

```
sudo /usr/local/bin/tcpreplay -i enp1s0f0 -M 9000 -K ~/pcap/p1p1.pcap
File Cache is enabled
Actual: 6000000 packets (6120000000 bytes) sent in 5.04 seconds.
Rated: 1124999172.7 Bps, 8999.99 Mbps, 1102940.36 pps
Flows: 1 flows, 0.18 fps, 6000000 flow packets, 0 non-flow
Statistics for network device: enp1s0f0
        Attempted packets:         6000000
        Successful packets:        6000000
        Failed packets:            0
        Truncated packets:         0
        Retried packets (ENOBUFS): 0
        Retried packets (EAGAIN):  0

sudo /usr/local/bin/tcpreplay -i qtx:enp1s0f0 -M 9000 -K ~/pcap/p1p1.pcap
File Cache is enabled
Actual: 6000000 packets (6120000000 bytes) sent in 13.04 seconds.
Rated: 456505641.7 Bps, 3652.04 Mbps, 447554.55 pps
Flows: 1 flows, 0.07 fps, 6000000 flow packets, 0 non-flow
Statistics for network device: /dev/net/quick_tx_enp1s0f0
        Attempted packets:         6000000
        Successful packets:        6000000
        Failed packets:            0
        Truncated packets:         0
        Retried packets (ENOBUFS): 0
        Retried packets (EAGAIN):  0
```

Looking the quick_tx source, I noticed they are only for e1000 and virtio, does it mean quick_tx is not formally supported with ixgbe?

**Comment by fklassen** [comment]
All adapters should be supported by quick-tx. I thought that we were getting full-wire rate with ixgbe. I'll mark this as a bug and try to dig into it.

**Comment by duanehoward** [comment]
I'm also seeing this with ixgbe. When trying to use qtx with ixgbe, dmesg fills up with errors like the following:
[72255.119287] [quick_tx] INFO:  Device registered: /dev/net/quick_tx_lo --> lo
[72255.119413] [quick_tx] INFO:  Device registered: /dev/net/quick_tx_eth4 --> eth4
[72255.119566] [quick_tx] INFO:  Device registered: /dev/net/quick_tx_eth5 --> eth5
[72255.119696] [quick_tx] INFO:  Device registered: /dev/net/quick_tx_eth0 --> eth0
[72255.119849] [quick_tx] INFO:  Device registered: /dev/net/quick_tx_eth1 --> eth1
[72255.119894] [quick_tx] INFO:  Device registered: /dev/net/quick_tx_bond0 --> bond0
[72309.152329] ------------[ cut here ]------------
[72309.152353] WARNING: CPU: 0 PID: 0 at /build/buildd/linux-lts-trusty-3.13.0/net/sched/sch_generic.c:264 dev_watchdog+0x267/0x270()
[72309.152356] NETDEV WATCHDOG: eth1 (ixgbe): transmit queue 0 timed out
[72309.152358] Modules linked in: quick_tx(OX) btrfs raid6_pq xor ufs qnx4 hfsplus hfs minix ntfs msdos jfs xfs libcrc32c reiserfs cpuid 8021q mrp garp bonding stp llc gpio_ich dcdbas sb_edac edac_core joydev shpchp lpc_ich mei_me ipmi_si mei ipmi_devintf wmi mac_hid acpi_power_meter lp parport binfmt_misc ixgbe igb hid_generic dca i2c_algo_bit usbhid ptp hid pps_core megaraid_sas mdio [last unloaded: pf_ring]
[72309.152427] CPU: 0 PID: 0 Comm: swapper/0 Tainted: G           OX 3.13.0-43-generic #72~precise1-Ubuntu
[72309.152429] Hardware name: Dell Inc. PowerEdge R720xd/0C4Y3R, BIOS 1.5.2 02/01/2013
[72309.152434]  0000000000000108 ffff880fffc03d28 ffffffff81757f91 0000000000003c66
[72309.152443]  ffff880fffc03d78 ffff880fffc03d68 ffffffff8106afcc ffff880fffc03d88
[72309.152449]  ffff880fe3d00000 ffff880fe3d00360 ffff880fe613cec0 0000000000000040
[72309.152456] Call Trace:
[72309.152470]  <IRQ>  [<ffffffff81757f91>] dump_stack+0x46/0x58
[72309.152490]  [<ffffffff8106afcc>] warn_slowpath_common+0x8c/0xc0
[72309.152496]  [<ffffffff8106b0b6>] warn_slowpath_fmt+0x46/0x50
[72309.152503]  [<ffffffff81671477>] dev_watchdog+0x267/0x270
[72309.152509]  [<ffffffff81085de0>] ? __queue_work+0x3c0/0x3c0
[72309.152514]  [<ffffffff81671210>] ? pfifo_fast_dequeue+0xe0/0xe0
[72309.152519]  [<ffffffff810784d5>] call_timer_fn+0x45/0x160
[72309.152524]  [<ffffffff81079890>] run_timer_softirq+0x280/0x300
[72309.152530]  [<ffffffff81045833>] ? lapic_next_deadline+0x33/0x40
[72309.152545]  [<ffffffff81671210>] ? pfifo_fast_dequeue+0xe0/0xe0
[72309.152561]  [<ffffffff8107029d>] __do_softirq+0xdd/0x300
[72309.152566]  [<ffffffff8107085e>] irq_exit+0x11e/0x140
[72309.152573]  [<ffffffff8176fa7a>] smp_apic_timer_interrupt+0x4a/0x60
[72309.152578]  [<ffffffff8176e3dd>] apic_timer_interrupt+0x6d/0x80
[72309.152579]  <EOI>  [<ffffffff815fd06e>] ? cpuidle_enter_state+0x5e/0xe0
[72309.152589]  [<ffffffff815fd067>] ? cpuidle_enter_state+0x57/0xe0
[72309.152594]  [<ffffffff815fd1b0>] cpuidle_idle_call+0xc0/0x210
[72309.152601]  [<ffffffff8101e9de>] arch_cpu_idle+0xe/0x30
[72309.152610]  [<ffffffff810c3da8>] cpu_idle_loop+0x78/0x270
[72309.152617]  [<ffffffff81d33120>] ? early_idt_handlers+0x120/0x120
[72309.152638]  [<ffffffff810c400b>] cpu_startup_entry+0x6b/0x70
[72309.152652]  [<ffffffff8173fbf7>] rest_init+0x77/0x80
[72309.152657]  [<ffffffff81d33fb4>] start_kernel+0x39e/0x3a5
[72309.152661]  [<ffffffff81d33aa8>] ? repair_env_string+0x5a/0x5a
[72309.152667]  [<ffffffff81746b1c>] ? memblock_reserve+0x46/0x4b
[72309.152671]  [<ffffffff81d335f8>] x86_64_start_reservations+0x2a/0x2c
[72309.152676]  [<ffffffff81d3373e>] x86_64_start_kernel+0x144/0x153
[72309.152679] ---[ end trace 867122ad9a1cae4e ]---
[72309.152685] ixgbe 0000:01:00.1 eth1: initiating reset due to tx timeout
[72309.152748] ixgbe 0000:01:00.1 eth1: Reset adapter
[72310.000627] ixgbe 0000:01:00.0 eth0: NIC Link is Down
[72310.693058] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72310.792972] ixgbe 0000:01:00.1 eth1: Reset adapter
[72312.617779] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72312.717693] ixgbe 0000:01:00.1 eth1: Reset adapter
[72314.654528] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72314.754538] ixgbe 0000:01:00.1 eth1: Reset adapter
[72315.877568] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72315.878909] ixgbe 0000:01:00.1 eth1: Reset adapter
[72317.003191] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72317.003310] ixgbe 0000:01:00.1 eth1: Reset adapter
[72319.028176] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72319.128084] ixgbe 0000:01:00.1 eth1: Reset adapter
[72321.052919] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72321.152847] ixgbe 0000:01:00.1 eth1: Reset adapter
[72322.773560] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72322.873482] ixgbe 0000:01:00.1 eth1: Reset adapter
[72323.982343] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72323.985885] ixgbe 0000:01:00.1 eth1: Reset adapter
[72325.096050] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72325.098299] ixgbe 0000:01:00.1 eth1: Reset adapter
[72326.707036] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72326.806956] ixgbe 0000:01:00.1 eth1: Reset adapter
[72328.727785] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72328.827700] ixgbe 0000:01:00.1 eth1: Reset adapter
[72330.748533] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72330.848463] ixgbe 0000:01:00.1 eth1: Reset adapter
[72332.757284] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72332.857199] ixgbe 0000:01:00.1 eth1: Reset adapter
[72334.790042] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72334.890056] ixgbe 0000:01:00.1 eth1: Reset adapter
[72335.997160] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72335.998362] ixgbe 0000:01:00.1 eth1: Reset adapter
[72337.106244] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72337.106835] ixgbe 0000:01:00.1 eth1: Reset adapter
[72338.691495] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72338.791422] ixgbe 0000:01:00.1 eth1: Reset adapter
[72340.712263] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72340.812198] ixgbe 0000:01:00.1 eth1: Reset adapter
[72342.733014] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72342.832920] ixgbe 0000:01:00.1 eth1: Reset adapter
[72344.761762] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72344.861672] ixgbe 0000:01:00.1 eth1: Reset adapter
[72346.786618] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72346.886448] ixgbe 0000:01:00.1 eth1: Reset adapter
[72348.711237] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72348.811148] ixgbe 0000:01:00.1 eth1: Reset adapter
[72349.923359] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72349.923552] ixgbe 0000:01:00.1 eth1: Reset adapter
[72351.033554] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72351.035983] ixgbe 0000:01:00.1 eth1: Reset adapter
[72352.150011] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72352.152395] ixgbe 0000:01:00.1 eth1: Reset adapter
[72353.262387] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72353.264819] ixgbe 0000:01:00.1 eth1: Reset adapter
[72354.364561] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72354.365218] ixgbe 0000:01:00.1 eth1: Reset adapter
[72355.466480] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72355.469650] ixgbe 0000:01:00.1 eth1: Reset adapter
[72356.578331] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72356.582057] ixgbe 0000:01:00.1 eth1: Reset adapter
[72357.687316] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72357.690469] ixgbe 0000:01:00.1 eth1: Reset adapter
[72358.803840] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72358.806876] ixgbe 0000:01:00.1 eth1: Reset adapter
[72359.915737] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72359.919305] ixgbe 0000:01:00.1 eth1: Reset adapter
[72361.022051] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72361.023710] ixgbe 0000:01:00.1 eth1: Reset adapter
[72362.130433] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72362.132142] ixgbe 0000:01:00.1 eth1: Reset adapter
[72363.242095] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72363.244559] ixgbe 0000:01:00.1 eth1: Reset adapter
[72364.351299] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72364.352940] ixgbe 0000:01:00.1 eth1: Reset adapter
[72365.463036] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72365.465382] ixgbe 0000:01:00.1 eth1: Reset adapter
[72366.574690] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72366.577774] ixgbe 0000:01:00.1 eth1: Reset adapter
[72367.687852] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72367.690197] ixgbe 0000:01:00.1 eth1: Reset adapter
[72368.797690] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72368.798606] ixgbe 0000:01:00.1 eth1: Reset adapter
[72369.908241] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72369.911015] ixgbe 0000:01:00.1 eth1: Reset adapter
[72371.020946] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72371.023433] ixgbe 0000:01:00.1 eth1: Reset adapter
[72372.130939] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72372.131874] ixgbe 0000:01:00.1 eth1: Reset adapter
[72373.241842] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72373.244306] ixgbe 0000:01:00.1 eth1: Reset adapter
[72374.353344] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72374.356683] ixgbe 0000:01:00.1 eth1: Reset adapter
[72375.470716] ixgbe 0000:01:00.1 eth1: initiating reset to clear Tx work after link loss
[72375.473103] ixgbe 0000:01:00.1 eth1: Reset adapter
[72375.873216] BUG: soft lockup - CPU#8 stuck for 23s! [kworker/u68:1:2087]

**Comment by fklassen** [comment]
@duanehoward

I am moving your issue to #171 and removing it from this thread. You are reporting a new bug.

**Comment by fklassen** [comment]
Dropping support for Quick TX until a maintainer becomes available #357
